### PR TITLE
Feature/http stream

### PR DESCRIPTION
**Pull request by crush-157**
Did a bit more refactoring.  Pulled format and listener out into separate methods.

### CODE REVIEW

File: lib/fdk/runner.rb [hunk 1]
```diff
@@ -98,9 +106,9 @@ def self.handle_function(function, req, resp)
 
     # TODO: gimme a bit me flexibility on response handling
```
Comment (denismakogon) [inline lib/fdk/runner.rb]: Would be nice to have the context split to InvokeContext and HTTPGateway context. At this moment the key difference is in headers, InvokeContext doesn't decap headers (decap from `Fn-Http-H-` prefix). But an HTTPGateway context does. Also, context attributes also changed, no more `FN_TYPE`, new headers:  `FN_ID`, etc.

Comment (zootalures) [reply lib/fdk/runner.rb]: It's done

Comment (zootalures) [reply lib/fdk/runner.rb]: see lib/fdk/context.rb HTTPContext

Comment (denismakogon) [reply lib/fdk/runner.rb]: I was looking for that on the master branch, but it's on the different one. Now i see that.

**Pull request merged**